llvm/include/llvm/Support/GenericDomTreeConstruction.h

```diff
@@ -552,7 +553,7 @@ struct SemiNCAInfo {
 
     addVirtualRoot();
     unsigned Num = 1;
-    for (const NodePtr Root : DT.Roots) Num = runDFS(Root, Num, DC, 0);
+    for (const NodePtr Root : DT.Roots) Num = runDFS(Root, Num, DC, 1);
```

Comment:
Maybe we can pass `Num` here to make it clear that we consider is to be the last one used *and* the node for attachment?

Reply:
I don't think that would be correct. Note that this code updates `Num` on each iteration. For the first root, both will be the same, but for the later one the attachment needs to stay at 1 while the last num increases.

Reply:
Ah yes yes.